aten/src/ATen/native/AveragePool2d.cpp

```diff
@@ -82,7 +82,7 @@ TORCH_PRECOMPUTE_META_FUNC(avg_pool2d)
         input.options().memory_format(memory_format));
   }
 
-  return {kH, kW, dH, dW, padH, padW};
+  return TORCH_PRECOMPUTE_STRUCT(avg_pool2d)().set_kH(kH).set_kW(kW).set_dH(dH).set_dW(dW).set_padH(padH).set_padW(padW);
```

Comment:
So the old `{kH, kW, dH, dW, padH, padW}` syntax still works here because the precompute struct is declared as a `struct`. I would like to disallow that; could do so by making the `struct` a `class` where everything is public?

Reply:
It might be sufficient to define a constructor, not sure